.github/workflows/package-macos.yml

```diff
@@ -0,0 +1,123 @@
+          mkdir -p "$POETRY_HOME"
+          curl -sSL https://install.python-poetry.org | python3 - --version=${{ env.poetry-version }}
+          echo "$POETRY_HOME/bin" >> "$GITHUB_PATH"
+
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # pin v3.0.2
+
+      - uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb  # pin v4.0.0
+        with:
+          python-version: ${{ env.python-version }}
+          cache: poetry
+
+      - name: Get wheel version
+        id: wheel-version
+        run: |
+          set -e
+          # Retrieve Parsec version and customize it if we are not building on a tag
+          # See https://www.python.org/dev/peps/pep-0440/#local-version-identifiers
+          if ([ "${{ github.ref_type }}" == "tag" ])
+          then
+            # Use tag name as version
+            VERSION=`git describe --tag`
+          else
+            set -x
+            # Use `<tag_name>+dev<commit_hash>` as version
+            VERSION=`sed -n 's/^__version__ = "\\(.*\\)"$/\\1/p' parsec/_version.py`
+            COMMIT=`git rev-parse --verify --short HEAD`
+            VERSION="$VERSION.$COMMIT"
+            # Customize version with tag
+            sed -i .back -e "s/__version__ = .*/__version__ = \"$VERSION\"/" parsec/_version.py
+            test $(grep -E 'version = ' -o pyproject.toml | wc -l) -eq 1  # Sanity check for the next sed
+            sed -i .back -e "s#^\\(version = \\).*\$#\\1'$VERSION'#" pyproject.toml
+          fi
+
+          echo "::warning title=Wheel version::$VERSION"
+          echo "::set-output name=version::$VERSION"
+
+      - name: Create output dir
+        run: mkdir dist
+
+      - name: Set file for wheel version
+        run: echo ${{ steps.wheel-version.outputs.version }} > dist/version
+
+      - name: Build wheel
+        run: python packaging/wheel/wheel_it.py . --output dist
+
+      - name: Debug dist dir
+        run: ls dist
+
+      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # pin v3.1.0
+        with:
+          name: ${{ runner.os }}-wheel
+          path: dist/
+
+
+  package-macos-build-app:
+    runs-on: macos-12
+    needs: package-macos-build-wheel
+    env:
+      WHEEL_VERSION: ${{ needs.package-macos-build-wheel.outputs.wheel-version }}
+    steps:
+      - uses: actions/setup-python@d09bd5e6005b175076f227b13d9730d56e9dcfcb  # pin v4.0.0
+        with:
+          python-version: ${{ env.python-version }}
+
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # pin v3.0.2
+
+      - uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # pin v3.0.0
+        with:
+          name: ${{ runner.os }}-wheel
+          path: dist
+
+      - name: Copy packaging script
+        run: cp -Rv ${{ github.workspace }}/packaging/macOS/* ${{ runner.temp }}
+
+      - name: Freeze Parsec
+        env:
+          WORKSPACE_DIR: ${{ github.workspace }}
```

Comment:
I think it would be better to have `freeze_program.py` set this env variable before running pyinstaller instead, otherwise we might end up with unexpected behavior when running `freeze_program.py` outside of the CI (given the `program_source` argument passed to freeze program is not respected when pyinstaller is called)

Reply:
done